packages/ledgerstate/utxo_dag.go

```diff
@@ -247,26 +267,18 @@ func (u *UTXODAG) Consumers(outputID OutputID) (cachedConsumers CachedConsumers)
 
 // LoadSnapshot creates a set of outputs in the UTXO-DAG, that are forming the genesis for future transactions.
 func (u *UTXODAG) LoadSnapshot(snapshot *Snapshot) {
-	// storing genesis tx and genesis output
-	// store TransactionMetadata
-	transactionMetadata := NewTransactionMetadata(GenesisTransactionID)
-	transactionMetadata.SetSolid(true)
-	transactionMetadata.SetBranchID(MasterBranchID)
-	transactionMetadata.SetFinalized(true)
-
-	(&CachedTransactionMetadata{CachedObject: u.transactionMetadataStorage.ComputeIfAbsent(GenesisTransactionID.Bytes(), func(key []byte) objectstorage.StorableObject {
-		transactionMetadata.Persist()
-		transactionMetadata.SetModified()
-		return transactionMetadata
-	})}).Release()
+	for txID, record := range snapshot.Transactions {
+		transaction := NewTransaction(record.Essence, record.UnlockBlocks)
+		cached, storedTx := u.transactionStorage.StoreIfAbsent(transaction)
 
-	for txID, essence := range snapshot.Transactions {
-		transaction := NewTransaction(essence, UnlockBlocks{NewReferenceUnlockBlock(0)})
-		cached, stored := u.transactionStorage.StoreIfAbsent(transaction)
-		if stored {
+		if storedTx {
 			cached.Release()
 		}
-		for _, output := range essence.outputs {
+
+		for i, output := range record.Essence.outputs {
+			if !record.UnspentOutputs[i] {
+				continue
+			}
```

Comment:
Now that we have different types of outputs, the way we store `AddressOutputMapping` is outdated here on line 288.
One output might have more addresses mapped to it, take a look at this:
From `packages/tangle/booker.go` line 247-267:
```
for _, output := range transaction.Essence().Outputs() {
		switch output.Type() {
		case ledgerstate.AliasOutputType:
			castedOutput := output.(*ledgerstate.AliasOutput)
			// if it is an origin alias output, we don't have the aliasaddress from the parsed bytes.
			// that happens in utxodag output booking, so we calculate the alias address here
			b.tangle.LedgerState.UTXODAG.StoreAddressOutputMapping(castedOutput.GetAliasAddress(), output.ID())
			b.tangle.LedgerState.UTXODAG.StoreAddressOutputMapping(castedOutput.GetStateAddress(), output.ID())
			if !castedOutput.IsSelfGoverned() {
				b.tangle.LedgerState.UTXODAG.StoreAddressOutputMapping(castedOutput.GetGoverningAddress(), output.ID())
			}
		case ledgerstate.ExtendedLockedOutputType:
			castedOutput := output.(*ledgerstate.ExtendedLockedOutput)
			if castedOutput.FallbackAddress() != nil {
				b.tangle.LedgerState.UTXODAG.StoreAddressOutputMapping(castedOutput.FallbackAddress(), output.ID())
			}
			b.tangle.LedgerState.UTXODAG.StoreAddressOutputMapping(output.Address(), output.ID())
		default:
			b.tangle.LedgerState.UTXODAG.StoreAddressOutputMapping(output.Address(), output.ID())
		}
	}
```

Reply:
done